### ISSUE TITLE
Location Should Not Be Hardcoded conflict with Bicep

### ISSUE THREAD

**Issue by lr90** [Azure/arm-ttk#347]
This is failing when using a module scoped to the subscription:

```
targetScope = 'subscription'

param workspaceId string

module diagnosticSettings './activity-log-diagnostics.bicep' = {
  name: 'DiagnosticSettings'
  scope: subscription()
  params: {
    workspaceId: workspaceId
  }
}
```
The property "location" is not allowed on objects of type "module". Permissible properties include "dependsOn".bicep(BCP038) so can not be added and the transpiled code results in:
```
  "resources": [
    {
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2019-10-01",
      "name": "DiagnosticSettings",
      "location": "[deployment().location]",
      "properties": {
        "expressionEvaluationOptions": {
          "scope": "inner"
        },
        "mode": "Incremental",
        "parameters": {
          "workspaceId": { ...
          }
        },
        "template": { ...
        }
      }
    }
  ],
```
Which fails the location test which I would prefer not to disable :-)

**Comment by bmoore-msft** [comment]
@lr90 - I've removed the targeted check for deployment().location for now, and have #346 for a proper fix...  lmk if you're still seeing issues with this sceanrio.

**Comment by bmoore-msft** [comment]
Check for the TODO in the test for the remaining work

**Comment by StartAutomating** [comment]
@bmoore-msft I'm having trouble getting a reliable repro for this, though it appears it's already been fixed by #346.  We may be able to close this off the board.

**Comment by bmoore-msft** [comment]
Here's a sample template:

```json
{
  "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "metadata": {
    "_generator": {
      "name": "bicep",
      "version": "0.4.1124.51302",
      "templateHash": "1282479610345940392"
    }
  },
  "parameters": {
    "location": {
      "type": "string",
      "defaultValue": "[deployment().location]"
    }
  },
  "resources": [
    {
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2020-10-01",
      "name": "nested",
      "location": "[deployment().location]",
      "properties": {
        "expressionEvaluationOptions": {
          "scope": "inner"
        },
        "mode": "Incremental",
        "parameters": {
          "location": {
            "value": "[parameters('location')]"
          }
        },
        "template": {
          "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
          "contentVersion": "1.0.0.0",
          "metadata": {
            "_generator": {
              "name": "bicep",
              "version": "0.4.1124.51302",
              "templateHash": "13055239406992524583"
            }
          },
          "parameters": {
            "location": {
              "type": "string"
            }
          },
          "resources": [
            {
              "type": "Microsoft.Resources/resourceGroups",
              "apiVersion": "2021-04-01",
              "name": "foo",
              "location": "[parameters('location')]"
            }
          ]
        }
      }
    }
  ]
}
```

And then you have to restore this check to repro it: https://github.com/Azure/arm-ttk/blob/master/arm-ttk/testcases/deploymentTemplate/Location-Should-Not-Be-Hardcoded.test.ps1#L68
